### PR TITLE
Add user following

### PR DESCRIPTION
**Pull request by piiknight**

### CODE REVIEW

**Comment by piiknight** [comment]
ready

File: app/models/relationship.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
+  belongs_to :followed, class_name: "User"
```
Comment (tranducquoc) [inline app/models/relationship.rb]: class_name: User.name

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,11 @@
 class User < ApplicationRecord
   has_many :microposts, dependent: :destroy
+  has_many :active_relationships, class_name:  "Relationship",
+    foreign_key: "follower_id", dependent: :destroy
+  has_many :passive_relationships, class_name:  "Relationship",
```
Comment (tranducquoc) [inline app/models/user.rb]: tương tự

File: app/views/relationships/create.js.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+$('#follow_form').html('<%= escape_javascript(render('users/unfollow')) %>');
```
Comment (tranducquoc) [inline app/views/relationships/create.js.erb]: escape_javascript => j

File: app/views/relationships/destroy.js.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+$('#follow_form').html('<%= escape_javascript(render('users/follow')) %>');
```
Comment (tranducquoc) [inline app/views/relationships/destroy.js.erb]: tương tự

**Comment by tranducquoc** [comment]
commented

**Comment by piiknight** [comment]
ready

**Comment by tranducquoc** [comment]
merged

**Pull request merged**